### PR TITLE
Server-side rendering and hydration prerequisites

### PR DESCRIPTION
**Pull request by dcsaszar**
The basic prerendering concepts are copied from https://github.com/Scrivito/scrivito_example_app_js

### CODE REVIEW

File: src/Objs/Product/ProductComponent.tsx [hunk 1]
```diff
@@ -192,7 +192,7 @@ const CartActionButton = connect(function CartActionButton({
   const cartLoginLabel = getMessage('cartLoginLabel')
   const cartUnvailableMessage = getMessage('cartUnvailableMessage')
 
-  if (!isUserLoggedIn()) {
+  if (typeof window === 'undefined' || !isUserLoggedIn()) {
```
Comment (apepper) [inline src/Objs/Product/ProductComponent.tsx:195]: Could you please reference issue #11101 for usages of `isUserLoggedIn()`?

File: src/main.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
```
Comment (apepper) [inline src/main.tsx:1]: There is one more usage of this in `ErrorBoundary.tsx`. Could you switch this to direct import as well?

File: src/main.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { StrictMode } from 'react'
-import ReactDOM from 'react-dom/client'
```
Comment (apepper) [inline src/main.tsx:2]: `scrivitoExtensions.tsx` uses `import * as ReactDOM from 'react-dom/client'`. Could you switch this to direct import as well?

**Review by apepper — COMMENTED** [review]
FEEDBACK ([6c06e3c](https://github.com/Scrivito/scrivito-portal-app/pull/520/commits/6c06e3c0583b4dbe5fc7a8bb564d3f66c28c9d2d))

Nice 🚀! I only have some nitpick comments.

File: src/config/scrivito.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { configure } from 'scrivito'
 import { baseUrlForSite, siteForUrl } from './scrivitoSites'
 import { scrivitoTenantId } from './scrivitoTenants'
 
-export function configureScrivito() {
+export function configureScrivito(options?: { priority?: 'background' }) {
```
[the remark this answers is not in the document]
Comment (dcsaszar) [reply src/config/scrivito.ts:5]: We've never had another requirement and I didn't want to step in YAGNI territory. https://github.com/search?q=repo%3AScrivito%2Fscrivito_example_app_js%20priority&type=code

**Review by apepper — APPROVED** [review]
LGTM ([9b1bc6e](https://github.com/Scrivito/scrivito-portal-app/pull/520/commits/9b1bc6efee7bd4bf0779b8ad352543960bbcf18e))

Nice 🚀!

**Pull request merged**